content/en/blog/2023/traffic-for-ambient-and-sidecar/index.md

```diff
@@ -0,0 +1,105 @@
+
+**(1) (2) (3) (4)** the sleep container sends a request to httpbin. The request is intercepted by iptables rules and directed to port 15001 of the sleep pod sidecar. Then, the sidecar handles the request and routing rules by following configurations distributed by the control plane. Next, the request traffic is forwarded to a specific httpbin pod in node B, identified by its ip address.
+
+**(5) (6)**  After the request is sent to the device pair (`veth httpbin <-> eth0 inside httpbin pod`), the request would be intercepted and forwarded to the `istioin` device on the node B where httpbin pod is running by following its iptables and route rules. The `istioin` device in the node B is a `[geneve](https://www.rfc-editor.org/rfc/rfc8926.html)` tunnel, and the other side of this tunnel is `pistioin` device which is inside the ztunnel pod of node B.
+
+**(7) (8)** After the request enters the `pistioin` device of the ztunnel pod, the request is intercepted and redirected through port 15008 of the ztunnel pod by the iptables rules of the ztunnel pod on node B.
+
+**(9)** The traffic getting into the port 15008 would be considered as a inbound request, then ztunnel will forward the request to the httpbin pod in the same node B.
+
+### [ sleep-> sidecar ] -> waypoint -> ztunnel -> helloworld network traffic path
+
+Comparing with the first part of the diagram, it's clear that the waypoint proxy is added in path **"[ sleep -> sidecar ] -> ztunnel -> httpbin"**. The Istio control plane has all information of service and configuration of the service mesh. When helloworld is configured with a waypoint proxy, the EDS configuration of helloworld service received by sidecar of sleep pod will be changed to the type of `envoy_internal_address`. This causes that the request traffic going through the sidecar to be forwarded to port 15008 of the waypoint proxy on node C via the `[HBONE](https://docs.google.com/document/d/1Ofqtxqzk-c_wn0EgAXjaJXDHB9KhDuLe-W3YGG67Y8g/edit)` protocol.
+
+Similar to sidecar, waypoint proxy is based on envoy and it will forward the request to the helloworld pod based on envoy L7 routing strategy.
+
+## Wrapping up
+
+Sidecar mode is significant to make Istio a great service mesh solution. However, it also causes some problems because app and sidecar containers must be in a pod together. Istio ambient mode implements communication among services through centralized proxies (ztunnel and waypoint). This mode provides greater flexibility and scalability, reduces resource consumption through not needing a sidecar for each pod in the mesh, and allows more precise configuration. Therefore, it's no doubt that ambient mode is the next evolution of Istio. Currently, ambient is still alpha and sidecar mode is still the recommended mode of Istio. The coexistence of sidecar and ambient modes may be last a very long time and users can choose what is best for them at the pace they desire.
```

Comment:
almost done!  Except the last change, we do not want to use personal pronouns, and we also want to let user know all the information, so I update the last change as below:

```
It's obvious that the coexistence of sidecar and ambient modes may be last a very long time, although the ambient mode is still in alpha stage and the sidecar mode is still the recommended mode of Istio, it will give users a more light-weight option of running and adopting the Istio service mesh as the ambient mode moves towards beta and future releases.
```